### ISSUE TITLE
postgresql cast error 

### ISSUE THREAD

**Issue by WingDust** [sqlfluff/sqlfluff#3952]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

```sql
create cast (varchar as integer) with inout [as implicit];
```
![image](https://user-images.githubusercontent.com/45911286/195092980-2d4dd3a1-efc9-4dd4-afc7-e1bbc966de04.png)

### Expected Behaviour

sqlfluff lint right

### Observed Behaviour

```sql
create cast (varchar as integer) with inout [as implicit];
```

### How to reproduce

```sql
create cast (varchar as integer) with inout [as implicit];
```

### Dialect

Found unparsable section: 'create cast (varchar as integer) with in...'

### Version

1.3.2

### Configuration

None

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by greg-finley** [comment]
Is it valid syntax?

```
gregoryfinley=# create cast (varchar as integer) with inout [as implicit];
ERROR:  syntax error at or near "["
LINE 1: create cast (varchar as integer) with inout [as implicit];
```

**Comment by WittierDinosaur** [comment]
Closing this as it isn't valid syntax, the [] around `as implicit` means it's optional, you shouldn't include them in the query
